ios/Exponent/Kernel/DevSupport/EXHomeModule.m

```diff
@@ -21,6 +21,144 @@ @implementation EXHomeModule
+    if (!binaryString) {
+      return nil;
+    }
+    NSScanner *scanner = [NSScanner scannerWithString:binaryString];
+    BOOL ok = [scanner scanUpToString:@"<plist" intoString:nil];
+    if (!ok) { NSLog(@"unable to find beginning of plist"); return nil; }
+    NSString *plistString;
+    ok = [scanner scanUpToString:@"</plist>" intoString:&plistString];
+    if (!ok) { NSLog(@"unable to find end of plist"); return nil; }
+    plistString = [NSString stringWithFormat:@"%@</plist>",plistString];
+    // juggle latin1 back to utf-8!
+    NSData *plistdata_latin1 = [plistString dataUsingEncoding:NSISOLatin1StringEncoding];
+    //    plistString = [NSString stringWithUTF8String:[plistdata_latin1 bytes]];
+    //    NSData *plistdata2_latin1 = [plistString dataUsingEncoding:NSISOLatin1StringEncoding];
+    NSError *error = nil;
+    mobileProvision = [NSPropertyListSerialization propertyListWithData:plistdata_latin1 options:NSPropertyListImmutable format:NULL error:&error];
+    if (error) {
+      NSLog(@"error parsing extracted plist — %@",error);
+      if (mobileProvision) {
+        mobileProvision = nil;
+      }
+      return nil;
+    }
+  }
+  return mobileProvision;
+}
+
+-(EXClientReleaseType) getClientReleaseType {
+  NSDictionary *mobileProvision = [self getMobileProvision];
+  if (!mobileProvision) {
+    // failure to read other than it simply not existing
+    return EXClientReleaseTypeUnknown;
+  } else if (![mobileProvision count]) {
+#if TARGET_IPHONE_SIMULATOR
+    return EXClientReleaseSimulator;
+#else
+    return EXClientReleaseAppStore;
+#endif
+  } else if ([[mobileProvision objectForKey:@"ProvisionsAllDevices"] boolValue]) {
+    // enterprise distribution contains ProvisionsAllDevices - true
+    return EXClientReleaseEnterprise;
+  } else if ([mobileProvision objectForKey:@"ProvisionedDevices"] && [[mobileProvision objectForKey:@"ProvisionedDevices"] count] > 0) {
+    // development contains UDIDs and get-task-allow is true
+    // ad hoc contains UDIDs and get-task-allow is false
+    NSDictionary *entitlements = [mobileProvision objectForKey:@"Entitlements"];
+    if ([[entitlements objectForKey:@"get-task-allow"] boolValue]) {
+      return EXClientReleaseDev;
+    } else {
+      return EXClientReleaseAdHoc;
+    }
+  } else {
+    // app store contains no UDIDs (if the file exists at all?)
+    return EXClientReleaseAppStore;
+  }
+}
+
+- (NSString *)clientReleaseTypeToJS: (EXClientReleaseType) releaseType
+{
+  switch (releaseType)
+  {
+    case EXClientReleaseTypeUnknown:
+      return @"UNKNOWN";
+    case EXClientReleaseSimulator:
+      return @"SIMULATOR";
+    case EXClientReleaseEnterprise:
+      return @"ENTERPRISE";
+    case EXClientReleaseDev:
+      return @"DEVELOPMENT";
+    case EXClientReleaseAdHoc:
+      return @"ADHOC";
+    case EXClientReleaseAppStore:
+      return @"APPLE_APP_STORE";
+  }
+}
+
+- (EXClientReleaseType) getClientReleaseTypeAsync
```

Comment:
this looks the same as `getClientReleaseType` and isn't async, perhaps we should get rid of `getClientReleaseTypeAsync`?